Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPISidebarCocoa.mm

```diff
@@ -176,19 +206,24 @@
         return;
     }
 
+    if (!hasActiveUserGesture(*tab)) {
+        completionHandler(toWebExtensionError(nil, nil, @"open can only be called in response to a user action"));
```

Comment:
i would spell out the full api name here as opposed to just "open". i think you could pass it in when you call this method from WebProcess

Reply:
Unfortunately since this services both `sidebarAction.open()` and `sidePanel.open()` it's hard to know which one is in use here. I was thinking about maybe trying to augment the error message on the WebProcess side, since it feels a tad excessive to me to push the API name through IPC just for error message generation.

Reply:
In this case you would put the function name in the first argument of `toWebExtensionError()`. We should do that! But in this case, the error would be in the web process with the correct gesture check so it does not matter here.

For the API name, you can add a helper that returns the right "sidePanel" or "sidebarAction" string.